config/job.js

```diff
@@ -60,13 +60,26 @@ const SCHEMA_STEP_OBJECT = Joi.object()
 const SCHEMA_STEP = Joi.alternatives().try(SCHEMA_STEP_STRING, SCHEMA_STEP_OBJECT);
 const SCHEMA_STEPS = Joi.array().items(SCHEMA_STEP).min(1);
 const SCHEMA_IMAGE = Joi.string();
+const SCHEMA_EMAIL = Joi.string().email();
+const SCHEMA_EMAILS = Joi.array()
+    .items(SCHEMA_EMAIL)
+    .min(0);
+const SCHEMA_NOTIFICATIONS = Joi.object()
+    .keys({
+        emails: SCHEMA_EMAILS
+    });
+const SCHEMA_SETTINGS = Joi.object()
+    .keys({
+        notifications: SCHEMA_NOTIFICATIONS
+    });
 const SCHEMA_JOB = Joi.object()
     .keys({
         steps: SCHEMA_STEPS,
         environment: SCHEMA_ENVIRONMENT,
         matrix: SCHEMA_MATRIX,
         image: SCHEMA_IMAGE,
-        secrets: SCHEMA_SECRETS
+        secrets: SCHEMA_SECRETS,
+        settings: SCHEMA_SETTINGS
```

Comment:
The more I think about this, the more we need to be careful about this name.  Maybe something like `features`.